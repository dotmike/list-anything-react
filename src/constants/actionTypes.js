export const USER_LOGGED_IN = "USER_LOGGED_IN";
export const USER_LOGGED_OUT = "USER_LOGGED_OUT";

export const SUBJECTS_FETCHED = "SUBJECTS_FETCHED";
export const SUBJECT_CREATED = "SUBJECT_CREATED";
export const SUBJECT_FETCHED = "SUBJECT_FETCHED";
export const SUBJECT_DELETED = "SUBJECT_DELETED";

export const SUBJECT_DATA_FETCHED = "SUBJECT_DATA_FETCHED";
export const SUBJECT_DATA_CREATED = "SUBJECT_DATA_CREATED";
export const SUBJECT_DATA_EDITED = "SUBJECT_DATA_EDITED";
export const SUBJECT_DATA_DELETED = "SUBJECT_DATA_DELETED";

export const SEARCH_RESULTS = "SEARCH_RESULTS";
