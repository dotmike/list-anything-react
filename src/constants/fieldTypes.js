export const TEXT_INPUT = "text_input";
export const URL_INPUT_IMG = "url_input_img";
export const NUMBER_INPUT = "number_input";
export const TEXTAREA = "textarea";
export const URL_INPUT = "url_input";
export const DATE_INPUT = "date_input";
export const TEXT_INPUT_LIST = "text_input_list";
export const URL_INPUT_LIST = "url_input_list";
