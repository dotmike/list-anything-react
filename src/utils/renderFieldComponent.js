import React from "react";

import * as fieldTypes from "../constants/fieldTypes";

import TextInputField from "../components/fields/TextInputField";
import UrlInputImgField from "../components/fields/UrlInputImgField";
import NumberInputField from "../components/fields/NumberInputField";
import TextareaField from "../components/fields/TextareaField";
import UrlInputField from "../components/fields/UrlInputField";
import DateInputField from "../components/fields/DateInputField";

export default fieldData => {
  const { field } = fieldData;
  switch (field.field_type) {
    case fieldTypes.TEXT_INPUT:
      return <TextInputField {...fieldData} />;
    case fieldTypes.URL_INPUT_IMG:
      return <UrlInputImgField {...fieldData} />;
    case fieldTypes.NUMBER_INPUT:
      return <NumberInputField {...fieldData} />;
    case fieldTypes.TEXTAREA:
      return <TextareaField {...fieldData} />;
    case fieldTypes.URL_INPUT:
      return <UrlInputField {...fieldData} />;
    case fieldTypes.DATE_INPUT:
      return <DateInputField {...fieldData} />;
    default:
      return <div key={field._id} />;
  }
};
