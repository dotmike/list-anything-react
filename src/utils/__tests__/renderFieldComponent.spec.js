import { shallow } from "enzyme";

import renderFieldComponent from "../renderFieldComponent";
import { TEXT_INPUT } from "../../constants/fieldTypes";

describe("renderFieldComponent", () => {
  describe("when passed a valid `field.field_type`", () => {
    const testData = {
      value: "testando...",
      editable: true,
      showFieldDescription: true,
      field: { _id: "1", description: "Testando:", field_type: TEXT_INPUT }
    };

    it("renders the correct field component", () => {
      const wrapper = shallow(renderFieldComponent(testData));
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when passed a valid `field.field_type`", () => {
    const testData = {
      field: { _id: "1", description: "Testando:", field_type: "test" }
    };

    it("renders the default div", () => {
      const wrapper = shallow(renderFieldComponent(testData));
      expect(wrapper).toMatchSnapshot();
    });
  });
});
