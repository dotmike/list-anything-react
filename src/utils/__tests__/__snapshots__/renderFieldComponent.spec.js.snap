// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderFieldComponent when passed a valid \`field.field_type\` renders the correct field component 1`] = `
<FormField
  error={false}
>
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    label="Testando:"
    name="1"
    onChange={[Function]}
    placeholder="Testando:"
    type="text"
    value="testando..."
  />
</FormField>
`;

exports[`renderFieldComponent when passed a valid \`field.field_type\` renders the default div 1`] = `
<div
  key="1"
/>
`;
